local.go

```diff
@@ -276,32 +277,41 @@ func checkConvergence(loc *Location, itertype IterationType, stats *Stats, setti
 	return NotTerminated
 }
 
-// evaluate evaluates the function and stores the answer in place
+// evaluate evaluates the function given by funcs at xNext, stores the answer
+// into location and updates stats. If location.X is not equal to xNext and
+// evalType == NoEvaluation, location is filled with NaN values. If the
```

Comment:
Things are filled with NaN values for lots of reasons. How about 

"If location.X is not equal to xNext, then unused fields of Location are set to NaN"